src/run.rs

```diff
@@ -147,11 +149,19 @@ impl WorkerSession {
                 self.trial_idx = 0;
                 Ok(flow)
             }
-            Err(err) if self.trial_idx >= self.context.max_retries_per_op => Err(err),
+            Err(err) if self.trial_idx >= self.context.max_retries_per_op => {
+                Err(err.context(format!(
+                    "Maximum number of retries per operation exceeded ({})",
+                    self.context.max_retries_per_op,
+                )))
+            }
```

Comment:
The argument that user passed talked about the maximum number of errors, so maybe it would be better to mention errors in the message as well.

Reply:
I didn't want to do it because the terminology used in the context talks about retries, not errors - however the error counts might make more sense to users, so I'll use that.